gooddata-java-model/src/main/java/com/gooddata/sdk/model/hierarchicalconfig/SettingItem.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2004-2021, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.sdk.model.hierarchicalconfig;
+
+import com.fasterxml.jackson.annotation.*;
+import com.gooddata.sdk.common.util.GoodDataToStringBuilder;
+import org.springframework.web.util.UriTemplate;
+
+
+import static com.gooddata.sdk.common.util.Validate.*;
+import static com.gooddata.sdk.model.hierarchicalconfig.SourceType.*;
+
+/**
+ * Contains information about hierarchical configuration object aka platform setting aka feature flag.
+ * Purpose of setting item is carrying information about some specific feature of Gooddata platform.
+ */
+@JsonTypeName("settingItem")
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SettingItem {
+
+    public static final UriTemplate CLIENT_SETTING_TEMPLATE = new UriTemplate(CLIENT.getApiUri() + "/{settingName}");
+    public static final UriTemplate DATA_PRODUCT_SETTING_TEMPLATE = new UriTemplate(DATA_PRODUCT.getApiUri() + "/{settingName}");
+    public static final UriTemplate SEGMENT_SETTING_TEMPLATE = new UriTemplate(SEGMENT.getApiUri() + "/{settingName}");
+    public static final UriTemplate DOMAIN_SETTING_TEMPLATE = new UriTemplate(DOMAIN.getApiUri() + "/{settingName}");
+    public static final UriTemplate PROJECT_SETTING_TEMPLATE = new UriTemplate(PROJECT.getApiUri() + "/{settingName}");
+    public static final UriTemplate PROJECT_GROUP_SETTING_TEMPLATE = new UriTemplate(PROJECT_GROUP.getApiUri() + "/{settingName}");
+
+    private final String key;
+    private final SourceType source;
+    private String value;
+
+    @JsonIgnore
+    private final Links links;
+
+    /**
+     * Creates new setting with given key, value and source.
+     *
+     * @param key unique name of setting
+     * @param value value of setting
+     * @param source source of setting
+     */
+    public SettingItem(String key, String value, String source) {
```

Comment:
I think that the source should not be in this constructor - it should be only present, when getting it from the platform.

Reply:
Removed